File: src/Services/EntertainmentService/Core/Entertainment.Domain/Entities/Book.cs
using Entertainment.Domain.Entities.Common;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace Entertainment.Domain.Entities
{
    public class Book : EntertainmentBase
    {
        public Book()
        {
            BookNotes = new List<BookNote>();
        }

        public string BookName { get; set; }
        public string AuthorName { get; set; }
        public int PageCount { get; set; }
        public bool IsFinished { get; set; } = false;

        [NotMapped]
        [JsonIgnore]
        public List<BookNote>? BookNotes { get; set; }
    }
}

File: src/WebApps/Angular/AdminClient/src/app/contracts/entertainment/book/book.ts
import { BookNote } from "../book-note/book_note";

export class Book {
    userId: String;
    createdDate: Date;
    updatedDate: Date;
    bookName: String;
    authorName: String;
    pageCount: number;
    isFinished: boolean;
    bookNotes: BookNote[] | undefined | null;
}